### PR TITLE
Integrate Go qps worker

### PR DESCRIPTION
**Pull request by jtattermusch**

### CODE REVIEW

File: tools/run_tests/performance/scenario_config.py [hunk 1]
```diff
@@ -522,11 +522,87 @@ def __str__(self):
...
+    return 600
+
+  def scenarios(self):
+    for secure in [True, False]:
+      secstr = 'secure' if secure else 'insecure'
+      smoketest_categories = [SMOKETEST] if secure else None
+
+      yield _ping_pong_scenario(
+          'go_generic_async_streaming_ping_pong_%s' % secstr, rpc_type='STREAMING',
+          client_type='ASYNC_CLIENT', server_type='ASYNC_GENERIC_SERVER',
```
Comment (vjpai) [inline tools/run_tests/performance/scenario_config.py]: I thought that async was not idiomatic in go. Is there a different interpretation of ASYNC_CLIENT, ASYNC_SERVER, and ASYNC_GENERIC_SERVER in go? @menghanl

**Comment by vjpai** [comment]
LGTM except for my question about Go, but that seems like a different category of question as it's about the worker itself. Would like input from the Go-folks.

**Comment by iamqizhao** [comment]
gRPC-Go does not have async mode. Here it simply starts a client/server regardless what type is requested. Let me know if you think getting an error/failure on async type is favored.

**Comment by vjpai** [comment]
So at some point in the future we'll add the option to return a failure. For now, I'm ok with knowing that it ignores the type. @jtattermusch - I think that means that there should be far fewer lines in this script, correct?

**Comment by jtattermusch** [comment]
@iamqizhao   should I change all the Go scenarios to use SYNC_CLIENT and SYNC_SERVER because that's the best fitting option for Go?

**Comment by jtattermusch** [comment]
I renamed the scenarios for Go to use "sync" instead of "async" wherever applicable.

**Comment by jtattermusch** [comment]
(sanity check failure is on master) and everything else seems to be green.

**Pull request merged**